dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/predictor/DefaultPredictionService.java

```diff
@@ -642,14 +642,14 @@ private DimensionalItemObject getPredictionReference( DataElement outputDataElem
     {
         for ( DimensionalItemObject item : sampleItems )
         {
-            if ( item == outputDataElement )
+            if ( item.equals( outputDataElement ) )
```

Comment:
Is this a standalone fix which should be applied individually and be backported?

Reply:
Good eye. As far as I know this is not a bug fix, but a more robust test for equality that is less prone to breaking in the future. This is in the code that sees if the output dataElement is also present in the sampled data, so values can be carried forward (for example, last month's balance + last month's change = this month's balance. This month's predicted balance is copied back into the input to predict next month's balance, and predictions are always made forward through time in Period order.) As far as I know the code works fine now, both in the unit test (where admittedly it will be the same data element object) and also in operational code. I could break this out into a separate PR if you think that's best, but I don't believe it is currently a bug.